pynumaflow/reducer/servicer/async_servicer.py

```diff
@@ -101,11 +103,14 @@ async def ReduceFn(
                     # append the task data to the existing task
                     # if the task does not exist, it will create a new task
                     await task_manager.append_task(request)
-        except Exception as e:
+        except BaseException as e:
             _LOGGER.critical("Reduce Error", exc_info=True)
-            context.set_code(grpc.StatusCode.UNKNOWN)
-            context.set_details(e.__str__())
-            raise e
+            # Send a context abort signal for the rpc, this is required for numa container to get
+            # the correct grpc error
+            await asyncio.gather(
+                context.abort(grpc.StatusCode.UNKNOWN, details=repr(e)), return_exceptions=True
+            )
+            terminate_on_stop(err=repr(e), parent=False)
```

Comment:
why not reuse/extend `exit_on_error`?

Reply:
`exit_on_error` internally calls `terminate_on_stop`
The difference being that `exit_on_error` manipulates the context before exiting. 
Here we were already aborting the context, hence I was calling terminate directly.

If required, I can extend the functionality to one function itself using a flag